core/include/jml/model.hpp

```diff
@@ -10,10 +10,38 @@
 
 namespace jml {
 
+class Model_Layer {
+  friend class Layer_Actv_Funcs_List;
+
+private:
+  Matrix* layer_matrix;
+  Vector* layer_bias_vector;
+  // Array of pointers to function pointers (because lonely function pointers might not be contiguous in memory)
+  // The size of this will be dictated by the associated matrix
+  double (*layer_actv_funcs_arr[])();
+};
+
+// Transition class for loading in/saving model, includes parameters and activation func ID's (enum <-> func pointer mappings?)
+class Model_Layer_Compressed {
+  friend class Model_Layer;
+
+};
+
 class Model {
+  friend class Model_Layer;
```

Comment:
Also, when the model is initialized, we should actually create the model layers. (And when the model is destroyed in `~Model`, it should free the model layers as well. This means passing in more data to the constructor, which we can tidy up in the future.